### PR TITLE
Update HikariCP and Paperlib

### PR DESCRIPTION
**Pull request by Narimm**
Update Hikaricp to 3.4.3 
  Changes in 3.4.3

    * Fixed 1534 check resultSet.getStatement() for null in ProxyDatabaseMetaData.

    * Add additional debug logging around connection add.

    * Add ThreadLocal removal attempt when bag item is removed.

   Changes in 3.4.2
    * fixed 1528 regression caused by pull request 1159 whereby fail-fast logic would exit upon all
      errors even if it was configured to continue trying.
Update Paperlib 1.0.3
  Async Chunks Urgent and Bed Spawn Location API

Signed-off-by: Narimm <benjicharlton@gmail.com>

### CODE REVIEW

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/commands/TeleportCommand.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069&bid=18048046)

File: src/main/java/me/botsko/prism/commands/TeleportCommand.java [hunk 1]
```diff
@@ -135,13 +132,25 @@ public void handle(CallInfo call) {
                         Prism.messenger.playerError("Action record occurred in world we can't find anymore."));
                 return;
             }
+            if (PaperLib.isPaper()) {
+                PaperLib.teleportAsync(call.getPlayer(), destinationAction.getLoc())
+                        .thenAccept(
+                              success -> sendTeleportCompleteMessage(success, call.getPlayer(), destinationAction));
+            } else {
+                sendTeleportCompleteMessage(call.getPlayer().teleport(destinationAction.getLoc()), call.getPlayer(),
```
Comment (alchemistmatt) [inline src/main/java/me/botsko/prism/commands/TeleportCommand.java:140]: Personally, I would split this into two statemets
```java
boolean success = call.getPlayer().teleport(destinationAction.getLoc());
sendTeleportCompleteMessage(success, call.getPlayer(), destinationAction);
```

**Review by alchemistmatt — COMMENTED** [review]
Code style suggestion, use this:
```java
boolean success = call.getPlayer().teleport(destinationAction.getLoc());
sendTeleportCompleteMessage(success, call.getPlayer(), destinationAction);
```

File: src/main/java/me/botsko/prism/commands/TeleportCommand.java (hunk 1 above)
Comment (Narimm) [reply src/main/java/me/botsko/prism/commands/TeleportCommand.java:140]: you cant
....because its aysnc in one case

Comment (Narimm) [reply src/main/java/me/botsko/prism/commands/TeleportCommand.java:140]: oh right you mean the last statement

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/commands/TeleportCommand.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069&bid=18048046)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/commands/TeleportCommand.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069&bid=18048046)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
- Added 1
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/commands/TeleportCommand.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069&bid=18048046)

File: src/main/java/me/botsko/prism/actions/ActionMeter.java [hunk 2]
```diff
@@ -17,41 +16,47 @@
...
     /**
      * Setup the meter to record.
-     *
-     * @param registry metric registry
      */
-    public static void setupActionMeter(MetricRegistry registry) {
-        for (Map.Entry<String, DripMeter> e : meter.entrySet()) {
-            registry.register(e.getKey(), e.getValue());
-        }
+    public static void setupActionMeter() {
```
Comment (Narimm) [inline src/main/java/me/botsko/prism/actions/ActionMeter.java:50]: Codacy Issue found: [Document empty method body](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069)

**Comment by Narimm** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 4
           

Complexity increasing per file
==============================
- src/main/java/me/botsko/prism/commands/TeleportCommand.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/AddstarMC/Prism-Bukkit/pullRequest?prid=5507069&bid=18048046)

**Pull request merged**